driver-core/src/test/java/com/datastax/driver/core/SessionErrorTest.java

```diff
@@ -84,14 +84,13 @@ public void should_not_propagate_unchecked_exceptions() {
             Session session = cluster.connect();
             // Pool to host1 should be still open because host1 is the control host,
             // but its pool should have no active connection
-            // Pool to host2 should have been closed because host2 has no
-            // more active connections
+            // Pool to host2 may be either closed because host2 was marked down before initialization
+            // was complete, or open and empty, if host2 was marked down after.
             Session.State state = session.getState();
```

Comment:
I think this is good, but alternatively it could be good to just wait for host2 to go down before checking the session state (`TestUtils.waitForDown("2", cluster);`).  This way we ensure that a side effect of the exception being raised is that the host is marked down and the pool is ultimately closed (the same assertions we had before can be maintained).